sdk/src/main/java/com/bugsnag/android/User.java

```diff
@@ -36,27 +31,131 @@ public void toStream(@NonNull JsonStream writer) throws IOException {
         writer.endObject();
     }
 
+    @Nullable
     public String getId() {
         return id;
     }
 
-    public void setId(String id) {
-        this.id = id;
-    }
-
+    @Nullable
     public String getEmail() {
         return email;
     }
 
-    public void setEmail(String email) {
-        this.email = email;
-    }
-
+    @Nullable
     public String getName() {
         return name;
     }
 
-    public void setName(String name) {
-        this.name = name;
+    /**
+     * Use this to create a new object holding user related details
+     *
+     * @return a builder for {@link User}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Create a new user object based on a previous user object.
+     *
+     * @param user the previous user object.
+     */
+    public static Builder builder(User user) {
+        return new Builder(user);
+    }
+
+    public static class Builder {
+        @Nullable String id;
+        @Nullable String email;
+        @Nullable String name;
+
+        public Builder() {
+        }
+
+        public Builder(User user) {
+            id(user.getId());
+            email(user.getEmail());
+            name(user.getName());
```

Comment:
A test case for null users would be useful here

Reply:
I agree, also found a mistake in `Error` that could lead to NPE, fixing that now.